docs/dep/dep-004.md

```diff
@@ -0,0 +1,76 @@
+using Helm. Many of the pipelines, if not all of them, share some tasks which causes Helm to try to
+deploy the same Task object for more than one pipelines. This is not allowed, since it would cause
+the same object to be either overwritten by a different package or it would cause it to be owned by
+multiple packages, which would break a lot of assumptions. We need to rethink how we manage our
+Tasks deployment.
+
+## Design
+
+Helm is the standard tool for managing Kuberenetes objects, and it has a lot of very useful
+features that would be difficult for us to replicate. It would be preferrable if we could keep
+using Helm, but in a more appropriate way. At the moment we are tightly coupling the definition of
+a workflow with the deployment of its resources. Specifically, we tend to produce a manifest that
+contains both the Tasks AND the Pipeline definition. However, Tasks and Pipelines are fundamentally
+different. Tasks contain image URLs, for example, or they could have various information related to
+the way that secrets and volumes or limits are applied in the cluster where the workflow is
+expected to run. A Pipeline definition should be the exact same between different cluster/
+environments/organisations since it's just a combination of Tasks.
+
+So our goal should be to modify and extend our code to have a clearer separation between the two
+phases and allow a finer control of the two stages.
+
+The first stage should be the deployment of the tasks or at least the discovery of the deployed
+tasks before a pipeline is deployed. The pipeline manifest should be just that: the pipeline
+manifest. That way we won't have to deal with Helm packages conflicting with each other due to the
+Tasks that they share.
+
+The Tasks themselves should be deployed using a Helm package, for example, all the Tasks in the
+Dracon repository could be aggregated in one package that is deployed everywhere. It is up to the
+administrator of the system to make sure that the correct Helm package is deployed and its
+variables are set correctly. Our system will not be responsible for that.
+
+On the other hand, tasks referenced in the pipeline definition using local paths or URLs will
+always be fetched and deployed or updated without checking things like version. It is up to the
+administrator of the system to make sure that the Task is in a state ready to be deployed.
+
+When a user wishes to refer to a Task deployed via Helm on the cluster, they will have to use a
+new URI type that is inspired by `pURL` and looks like this:
+`helm://{{ helm_package_id }}/name`
+
+The `helm_package_id` is the name of the Helm release deployed on the cluster and will be used to
+list the configmaps of the namespace and the Tasks using the label with name
+`app.kubernetes.io/instance` and value the `helm_package_id`. The system will expect to find a
+configmap with the label `v1.dracon.ocurity.com/bill-of-materials: tekton-tasks`. The configmap
```

Comment:
i don't understand why a configmap makes sense in this particular scenario of open source components?
shouldn't we instead have either `draconctl` listing the tasks installed in a cluster or the available tasks in a repository or OCI registry?

Reply:
I've significantly updated the RFC and removed this part after our offline discussion to simplify the structure of the system.